documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -81,7 +81,19 @@ the buffer were created for the same queue. For example:
     return 0;
   }
 
-Alternatively, use ``std::vector`` with a USM allocator. For example:
+Alternatively, use ``std::vector`` with a USM allocator.
+
+Note: The ability to appropriately detect USM allocated ``std::vector::iterator`` depends
+on details of the C++ standard library implementation and what information about the
+allocator is included in the ``std::vector::iterator`` type definition. If USM allocated
+vector iterators are not detectable with your C++ standard library, they will still function
+as inputs to oneDPL, but they will be treated as if they were host-side
+``std::vector::iterator`` as described in the section below. To guarantee no additional
+host-side copies, you can use ``std::vector::data()`` in combination with ``std::vector::size()``
+with USM allocated vectors to obtain a USM pointers. This will avoid extra host-side copies
```

Comment:
The tenses don't match in this "...USM allocated vectors to obtain a USM pointers."

I suggest either, depending on if you are talking about a single pointer or multiple:

- USM allocated vectors to obtain a USM pointer.
- USM allocated vectors to obtain USM pointers.

Reply:
I added a little extra to be more specific about what is being suggested, but it should resolve this issue.

Reply:
LGTM, aside from a comma instance (captured in a separate comment).